### PR TITLE
post-resolution issue changes + add status category

### PR DESCRIPTION
**Pull request by fivetran-jamie**
**What change(s) does this PR introduce?** 

- Allows for resolved issues that are updated after being resolved to have these changes reflected in daily history models
- add status category
- add incremental configs for Databricks -> the bulk of circle ci work is being done in https://github.com/fivetran/dbt_jira/pull/31https://github.com/fivetran/dbt_jira/pull/31

**Does this PR introduce a breaking change?**

- [ ] Yes (please provide breaking change details below.)
- [x] No  (please provide explanation how the change is non breaking below.)

**Is this PR in response to a previously created Issue**

- [x] Yes, Issue https://github.com/fivetran/dbt_jira/issues/29 and https://github.com/fivetran/dbt_jira/issues/22 (just the status category part -- versions will be in another PR or later in this PR...)
- [ ] No 

**How did you test the PR changes?** 

 
- [x] CircleCi  
- [ ] Other (please provide additional testing details below)
Also did a lot of running a full refresh with the old code, adding the new code, doing an incremental run with the new code (also with the old to compare), seeing if the missing days got backfilled

**Select which warehouse(s) were used to test the PR**
 
- [x] BigQuery
- [x] Redshift
- [x] Snowflake
- [ ] Postgres
- [ ] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
🏓 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

**Comment by fivetran-jamie** [comment]
@fivetran-joemarkiewicz would like to talk this though a little in standup

File: dbt_project.yml
Comment (fivetran-joemarkiewicz) [inline dbt_project.yml:2]: Would this not be a breaking change? If an existing user installs this release then their incremental runs will fail. We should upgrade this to `v0.4.0` to account for the breaking change.

File: integration_tests/dbt_project.yml
Comment (fivetran-joemarkiewicz) [inline integration_tests/dbt_project.yml:2]: Same note on upgrading to `v0.4.0`

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM after our discussion last week on the incremental logic changes! 

Some things we should probably do before release:
- Based off my code comment, we should upgrade this release to be `v0.4.0` as this does constitute a breaking change for users
- If the issue creator does not respond by the end of the week it would be worth reaching out to them via dbt Slack as they are pretty active in that community
- Wait to release a new version of this package until the end of this sprint to see if you are able to tackle the other issue attached to this package in this same `v0.4.0` release
- Be sure to add a note in the `v0.4.0` release that in order for these changes to take effect a `dbt run --full-refresh` will be required.

**Comment by fivetran-jamie** [comment]
moving postgres + spark compatibility to a [different PR](https://github.com/fivetran/dbt_jira/pull/31) since the customer will hopefully be trying this branch out to test the calendar spine stuff (and it was the same person who wants `versions` added, so combining these two tasks here)

**Comment by fivetran-jamie** [comment]
todo: add version logic here -- it's a little more complicated than i thought since versions are a multiselect issue-field

**Comment by fivetran-jamie** [comment]
- [ ] reminder to myself to re-generate documentation before merging

File: .circleci/config.yml
Comment (fivetran-joemarkiewicz) [inline .circleci/config.yml:51]: You should probably remove this before merging our else there will be merge conflicts with the PR you created that handles the postgres and spark compatibility.

File: packages.yml
Comment (fivetran-joemarkiewicz) [inline packages.yml:6]: Reminder to switch before merge

File: models/intermediate/int_jira__issue_versions.sql
Comment (fivetran-joemarkiewicz) [inline models/intermediate/int_jira__issue_versions.sql:18]: Can you explain these a bit more during standup? I want to make sure we aren't missing anything else.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM - Just a few small questions and comments that we should discuss before merging. It would also be helpful if we can run through the final results together during standup to help understand the changes a bit more.

File: models/intermediate/int_jira__issue_versions.sql
Comment (fivetran-jamie) [reply models/intermediate/int_jira__issue_versions.sql:18]: yeah these docs may help as well https://confluence.atlassian.com/jirakb/general-use-of-affect-version-s-and-fix-version-s-838428168.html

**Pull request merged**
